apps/dev-api/domains/miniapp/schema/AllB2CAppPropertiesService.test.js

```diff
@@ -0,0 +1,248 @@
+            let [app] = await createTestB2CApp(user)
+            await createTestB2CAppPublishRequest(support, app, {
+                isAppTested: true,
+                isContractSigned: true,
+                isInfoApproved: true,
+                status: PUBLISH_REQUEST_APPROVED_STATUS,
+            })
+            await publishB2CAppByTestClient(user, app)
+            app = await B2CApp.getOne(user, { id: app.id })
+            await updateCondoB2CApp(condoAdmin, { id: app.developmentExportId }, { importId: null, importRemoteSystem: null })
+            await publishB2CAppByTestClient(user, app, { info: true }, PROD_ENVIRONMENT)
+            app = await B2CApp.getOne(user, { id: app.id })
+            await updateCondoB2CApp(condoAdmin, { id: app.productionExportId }, { importId: null, importRemoteSystem: null })
+
+            const [devResult] = await allB2CAppPropertiesByTestClient(user, app, DEV_ENVIRONMENT)
+            expect(devResult).toHaveProperty(['meta', 'count'], 0)
+            const [prodResult] = await allB2CAppPropertiesByTestClient(user, app, PROD_ENVIRONMENT)
+            expect(prodResult).toHaveProperty(['meta', 'count'], 0)
+
+            await updateCondoB2CApp(condoAdmin, { id: app.developmentExportId }, {
+                deletedAt: dayjs().toISOString(),
+            })
+
+            await expectToThrowGQLError(async () => {
+                await allB2CAppPropertiesByTestClient(user, app, DEV_ENVIRONMENT)
+            }, {
+                code: BAD_USER_INPUT,
+                type: APP_NOT_FOUND,
+            }, 'result')
+            const [secondProdResult] = await allB2CAppPropertiesByTestClient(user, app, PROD_ENVIRONMENT)
+            expect(secondProdResult).toHaveProperty(['meta', 'count'], 0)
+
+            await updateCondoB2CApp(condoAdmin, { id: app.productionExportId }, {
+                deletedAt: dayjs().toISOString(),
+            })
+            await expectToThrowGQLError(async () => {
+                await allB2CAppPropertiesByTestClient(user, app, DEV_ENVIRONMENT)
+            }, {
+                code: BAD_USER_INPUT,
+                type: APP_NOT_FOUND,
+            }, 'result')
+            await expectToThrowGQLError(async () => {
+                await allB2CAppPropertiesByTestClient(user, app, PROD_ENVIRONMENT)
+            }, {
+                code: BAD_USER_INPUT,
+                type: APP_NOT_FOUND,
+            }, 'result')
+        })
+        test('Pagination must work', async () => {
+            const [firstDevResult] = await allB2CAppPropertiesByTestClient(user, app, DEV_ENVIRONMENT, {
+                first: 20,
+                skip: 0,
+            })
+            expect(firstDevResult.objs).toHaveLength(20)
+            const [secondDevResult] = await allB2CAppPropertiesByTestClient(user, app, DEV_ENVIRONMENT, {
+                first: 20,
+                skip: 20,
+            })
+            expect(secondDevResult.objs).toHaveLength(15)
```

Comment:
There's a lot of pagination tests. That's why I want to left it as is for now. It's more unclear for me to read `(const1 - const2)` instead of `15`. Cause when reading testsuite I can think the following:
1. Ok, we have 35 dev and prod properties as a baseline
2. Now lets test a pagination and correct returns of that 35.
3. So, now you're explicitly now, how to test out of ranges (skip 40), negatives, pagination and so on
4. But if I move it to const, I need to write a formulas for all this things, which decreases readability

Reply:
I read test: 
1. first: 20, 
2. skip: 20, 
3. result: 15. 
Why 15? Oh yes, somewhere at the top it says that there are 35 of them in total))